crates/matrix-sdk-ui/tests/integration/timeline/thread.rs

```diff
@@ -0,0 +1,308 @@
+    assert_pending!(timeline_stream);
+}
+
+#[async_test]
+async fn test_simple_thread() {
+    let server = MatrixMockServer::new().await;
+    let client = server.client_builder().build().await;
+
+    let room_id = room_id!("!a:b.c");
+    let sender_id = user_id!("@alice:b.c");
+
+    let factory = EventFactory::new().room(room_id).sender(sender_id);
+
+    let thread_root_event_id = owned_event_id!("$root");
+
+    server
+        .mock_room_event()
+        .match_event_id()
+        .ok(factory
+            .text_msg("Thread root")
+            .sender(sender_id)
+            .event_id(&thread_root_event_id)
+            .into())
+        .mock_once()
+        .mount()
+        .await;
+
+    let batch1 =
+        vec![factory.text_msg("Threaded event 2").event_id(event_id!("$2")).into_raw_sync().cast()];
+    let batch2 =
+        vec![factory.text_msg("Threaded event 1").event_id(event_id!("$1")).into_raw_sync().cast()];
+
+    server
+        .mock_room_relations()
+        .match_target_event(thread_root_event_id.clone())
+        .ok(RoomRelationsResponseTemplate::default().events(batch1).next_batch("next_batch"))
+        .mock_once()
+        .mount()
+        .await;
+
+    server
+        .mock_room_relations()
+        .match_target_event(thread_root_event_id.clone())
+        .match_from("next_batch")
+        .ok(RoomRelationsResponseTemplate::default().events(batch2))
+        .mock_once()
+        .mount()
+        .await;
+
+    let room = server.sync_joined_room(&client, room_id).await;
+
+    let timeline = Timeline::builder(&room)
+        .with_focus(TimelineFocus::Thread { root_event_id: thread_root_event_id, num_events: 1 })
+        .build()
+        .await
+        .unwrap();
+
+    let (items, mut timeline_stream) = timeline.subscribe().await;
+
+    assert_eq!(items.len(), 1 + 1); //  a date divider + the event
+    assert!(items[0].is_date_divider());
+    assert_eq!(
+        items[1].as_event().unwrap().content().as_message().unwrap().body(),
+        "Threaded event 2"
+    );
+    assert_pending!(timeline_stream);
+
+    let hit_start = timeline.paginate_backwards(1).await.unwrap();
+    assert!(hit_start);
+
+    assert_let!(Some(timeline_updates) = timeline_stream.next().await);
+
+    // Remove date separator and insert a new one plus the remaining threaded
+    // even and the thread root
+    assert_eq!(timeline_updates.len(), 4);
+
+    let items = timeline.items().await;
+    assert_eq!(items.len(), 4); // a date separator and the 3 events
+
+    assert!(items[0].is_date_divider());
+
+    assert_eq!(items[1].as_event().unwrap().content().as_message().unwrap().body(), "Thread root");
+
+    assert_eq!(
+        items[2].as_event().unwrap().content().as_message().unwrap().body(),
+        "Threaded event 1"
+    );
+
+    assert_eq!(
+        items[3].as_event().unwrap().content().as_message().unwrap().body(),
+        "Threaded event 2"
+    );
+}
+
+#[async_test]
+async fn test_thread_ordering() {
```

Comment:
Sorry, but what does this test do than the other isn't doing? Seems like it could be fused with the previous one. Or if it does something more, can you add a small comment at the top of the test's body, indicating what we're planning to test here, in a sentence or two?

Reply:
Coming back to it yeah, they can definitely be merged together. The idea was for the order one to focus or just how messages are ordered in the thread, skipping a lot of the other checks.